### ISSUE TITLE
uninitialized constant ActionView::PartialIteration on Rails 6.1.0.alpha

### ISSUE THREAD

**Issue by kayamy** [github/view_component#461]
Hey guys,

My collection components stopped working after upgrading from 6.0 to rails master. Some quick digging shows that `ActionView::PartialIteration` was moved between stable and edge.

It was moved here -> https://github.com/rails/rails/blob/master/actionview/lib/action_view/renderer/collection_renderer.rb

**stable** -> https://github.com/rails/rails/blob/6-0-stable/actionview/lib/action_view/renderer/partial_renderer.rb
**master** -> https://github.com/rails/rails/blob/master/actionview/lib/action_view/renderer/partial_renderer.rb

It's referenced over here in `view_component` -> https://github.com/github/view_component/blob/master/lib/view_component/collection.rb#L6

https://github.com/github/view_component/pull/306 seems related

I'm new to ruby and the entire rails ecosystem, so sorry if this is amateurish but I was able to fix it by copying the `PartialIteration` class into the `ViewComponent` module like in this fork https://github.com/fullpointer/view_component/commit/65f38dfb81f45bc4a59d50f3353ab0509e2446f5

Thanks!

### Steps to reproduce

- upgrade to rails 6.1.0.alpha
- render any any component using `with_collection` like so`<%= render CollaboratorComponent.with_collection(@collaborators) %>`

### System configuration
**Rails version**:
6.1.0.alpha
**Ruby version**:
2.7.1
**Gem version**:
2.18.1

**Comment by joelhawksley** [comment]
@fullpointer thanks for the report! Could you provided a PR with a failing test? It looks like our current suite passes with the latest Rails `master`: https://github.com/github/view_component/pull/464

**Comment by fullpointer** [comment]
I'll give it a try and report back!

**Comment by fullpointer** [comment]
@joelhawksley so i've spent a few hours trying to get the tests to fail without luck. Everything I try passes so it would indicate some issue with how the module is loaded. Forgive me if this is out of scope but I would really appreciate some guidance.

I found out that putting this line in `application.rb` forces `ActionView::PartialIteration` to show up:
`ActionView.constants.select { |c| ActionView.const_get(c).is_a? Class }` so I'm guessing this has to do with the lazy loading or order of things.

I also tried a bunch of different orders for requiring `action_cable/engine` in my `application.rb`. This is what it looks like currently.

```
require_relative "boot"

require "rails"
require "active_model/railtie"
require "active_job/railtie"
require "active_record/railtie"
require "active_storage/engine"
require "action_controller/railtie"
require "action_mailer/railtie"
require "action_mailbox/engine"
require "action_text/engine"
require "action_view/railtie"
require "view_component/engine"
require "action_cable/engine"
require "rails/test_unit/railtie"
require "clearance"
```

I'm not familiar with the ins and outs of how Rails loads things but I think the difference is that it gets loaded properly in the tests but not in my project.

I appreciate the help so far :)
